commit_test.go

```diff
@@ -62,6 +63,47 @@ func (s *SuiteCommit) TestParents(c *C) {
 	c.Assert(output, DeepEquals, expected)
 }
 
+func (s *SuiteCommit) TestCommitEncodeDecodeIdempotent(c *C) {
+	ts, err := time.Parse(time.RFC3339, "2006-01-02T15:04:05-07:00")
+	c.Assert(err, IsNil)
+	commits := []*Commit{
+		&Commit{
+			Author:    Signature{Name: "Foo", Email: "foo@example.local", When: ts},
+			Committer: Signature{Name: "Bar", Email: "bar@example.local", When: ts},
+			Message:   "Message\n\nFoo\nBar\nBaz\n\n",
+			tree:      core.NewHash("b029517f6300c2da0f4b651b8642506cd6aaf45d"),
+			parents:   []core.Hash{core.NewHash("b029517f6300c2da0f4b651b8642506cd6aaf45d")},
+		},
+		&Commit{
+			Author:    Signature{Name: "Foo", Email: "foo@example.local", When: ts},
+			Committer: Signature{Name: "Bar", Email: "bar@example.local", When: ts},
+			Message:   "Message\n\nFoo\nBar\nBaz\n\n",
+			tree:      core.NewHash("b029517f6300c2da0f4b651b8642506cd6aaf45d"),
+			parents: []core.Hash{
+				core.NewHash("b029517f6300c2da0f4b651b8642506cd6aaf45d"),
+				core.NewHash("c029517f6300c2da0f4b651b8642506cd6aaf45d"),
+				core.NewHash("d029517f6300c2da0f4b651b8642506cd6aaf45d"),
+				core.NewHash("e029517f6300c2da0f4b651b8642506cd6aaf45d"),
+			},
+		},
+	}
+	for _, commit := range commits {
+		// Compute hash
+		obj := &core.MemoryObject{}
+		err = commit.Encode(obj)
+		c.Assert(err, IsNil)
+		commit.Hash = obj.Hash()
```

Comment:
I am not sure I understand the purpose of 91-95, but it looks as if a bug in encode will make this test pass when it shouldn't.

will it be safer to precompute the hash as a fixture instead of relaying in commit.Encode to generate it?